src/main/java/uk/gov/hmcts/dts/fact/services/admin/AdminCourtGeneralInfoService.java

```diff
@@ -0,0 +1,47 @@
+package uk.gov.hmcts.dts.fact.services.admin;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.dts.fact.config.security.RolesProvider;
+import uk.gov.hmcts.dts.fact.entity.Court;
+import uk.gov.hmcts.dts.fact.exception.NotFoundException;
+import uk.gov.hmcts.dts.fact.model.admin.CourtGeneralInfo;
+import uk.gov.hmcts.dts.fact.repositories.CourtRepository;
+
+import static uk.gov.hmcts.dts.fact.services.admin.AdminRole.FACT_SUPER_ADMIN;
+
+@Service
+public class AdminCourtGeneralInfoService {
+    private final CourtRepository courtRepository;
+    private final RolesProvider rolesProvider;
+
+    @Autowired
+    public AdminCourtGeneralInfoService(final CourtRepository courtRepository, final RolesProvider rolesProvider) {
+        this.courtRepository = courtRepository;
+        this.rolesProvider = rolesProvider;
+    }
+
+    public CourtGeneralInfo getCourtGeneralInfoBySlug(final String slug) {
+        return courtRepository.findBySlug(slug)
+            .map(CourtGeneralInfo::new)
+            .orElseThrow(() -> new NotFoundException(slug));
+    }
+
+    public CourtGeneralInfo updateCourtGeneralInfo(final String slug, final CourtGeneralInfo generalInfo) {
+        final Court courtEntity = courtRepository.findBySlug(slug)
+            .orElseThrow(() -> new NotFoundException(slug));
+
+        courtEntity.setAlert(generalInfo.getAlert());
+        courtEntity.setAlertCy(generalInfo.getAlertCy());
+
+        if (rolesProvider.getRoles().contains(FACT_SUPER_ADMIN)) {
```

Comment:
If the role does not contain the super admin, should we return an error, or something indicating this? 

And also, it might be useful to put in some loggers to support this or enhance the debugging fun times if we need to do that in the higher environments

Reply:
If the role is not "superadmin" but it is "admin", you can set alert on lines 34-35 but not setting the rest of the information. Don't ask me why superadmin can set more information which to me is not sensitive information....it has always been like that so just do the same here :-) If it is NOT an admin or superadmin, the controller will return Unauthorised response and not even getting in here so I don't think logging is neccessary.